### PR TITLE
Update ts Locator definitions

### PR DESCRIPTION
**Pull request by LukoyanovE**
Updated `ILocator` definition (not founded use of `value` and removed it, added `id`)

Simplify TS definitions for functions with `locator`.

Fixed tests for parser.
Added tests for locator constructor (not necessary, but why not).

Need some TS developers review for errors, style and improvements

### CODE REVIEW

**Comment by LukoyanovE** [comment]
maybe `ilocator` should be changed to something like:
```
type ILocator =
  { id: string } |
  { xpath: string } |
  { css: string } |
  { name: string } |
  { frame: string } |
  { android: string } |
  { ios: string } |
  { react: string, props?: {}, state?: {} }
```

**Comment by DavertMik** [comment]
I'm approving, waiting for others to review.

File: lib/command/definitions.js [hunk 1]
```diff
@@ -123,14 +119,10 @@ declare function BeforeSuite(callback: ICodeceptCallback): void;
...
-declare function locate(selector: ILocator): Locator;
-declare function within(selector: string, callback: Function): Promise<any>;
-declare function within(selector: ILocator, callback: Function): Promise<any>;
-declare function session(selector: string, callback: Function): Promise<any>;
-declare function session(selector: ILocator, callback: Function): Promise<any>;
-declare function session(selector: string, config: any, callback: Function): Promise<any>;
-declare function session(selector: ILocator, config: any, callback: Function): Promise<any>;
+declare function locate(selector: LocatorOrString): Locator;
+declare function within(selector: LocatorOrString, callback: Function): Promise<any>;
+declare function session(selector: LocatorOrString, callback: Function): Promise<any>;
```
Comment (tsuemura) [inline lib/command/definitions.js]: Is `session` only taking a string as an first argument?

Comment (LukoyanovE) [reply lib/command/definitions.js]: Look like You are right.
Don't see any usage of `session` except with string

**Pull request merged**